types/connect-redis/index.d.ts

```diff
@@ -21,7 +24,7 @@ declare module "connect-redis" {
             new (options: RedisStoreOptions): session.Store;
         }
         interface RedisStoreOptions {
-            client?: redis.RedisClient;
+            client?: redis.RedisClient | ioredis.Redis;
```

Comment:
I would say that the correct thing to do here is create an interface for the common parts of `redis.RedisClient` and `ioredis.Redis` within the `redis` library, us that interface here (rather than the exact class type) and ensure that `ioredis` conforms to it, rather than using a union. It doesn't make sense for all `connect-redis` types to depend on `ioredis` when it's likely not in use.

Reply:
It makes sense making a "generic" type, but on the other side, the "client" value is just used once on initiation. All "client" operations are internal to the library.

I'm new to typescript, can I define a "wildcard" interface that makes sure a that the variable of different interface implements specific functions|properties|etc?

Reply:
That's precisely what an interface would be - TS is a structural language, so it exactly describes a set of properties/functions/etc which must be present.